ext/openssl/openssl.c

```diff
@@ -474,6 +474,47 @@ void php_openssl_store_errors()
 }
 /* }}} */
 
+/* {{{ php_openssl_errors_set_mark */
+void php_openssl_errors_set_mark(void) {
+	struct php_openssl_errors *errors;
+	struct php_openssl_errors *errors_mark;
+
+	if (!OPENSSL_G(errors)) {
+		return;
+	}
+
+	if (!OPENSSL_G(errors_mark)) {
+		OPENSSL_G(errors_mark) = pecalloc(1, sizeof(struct php_openssl_errors), 1);
+	}
+
+	errors = OPENSSL_G(errors);
+	errors_mark = OPENSSL_G(errors_mark);
+
+	memcpy(errors_mark, errors, sizeof(struct php_openssl_errors));
+}
+/* }}} */
+
+/* {{{ php_openssl_errors_restore_mark */
+void php_openssl_errors_restore_mark(void) {
+	struct php_openssl_errors *errors;
```

Comment:
we no longer require definitions at the top as at least C99 is used so just define it when first used (line 506).